dashboard/app/models/lesson.rb

```diff
@@ -442,18 +442,23 @@ def lockable_state(students)
     end
     return students.map do |student|
       user_level = student.last_attempt_for_any script_level.levels, script_id: script.id
-      # user_level_data is provided so that we can get back to our user_level when updating. in some cases we
-      # don't yet have a user_level, and need to provide enough data to create one
+      # user_level_data is provided so that we can get back to our user_level
+      # when updating. in some cases we don't yet have a user_level, and need
+      # to provide enough data to create one
+
+      # if we don't have a user level, consider ourselves locked
+      locked = user_level.nil? || user_level.show_as_locked?(self)
+      # if we don't have a user level, we can't be readonly
+      readonly = user_level.present? && user_level.show_as_readonly?(self, locked)
```

Comment:
Unfortunately, I don't think `present?` is the same as `! .nil?`.  (I've wanted a not_nil? method too!)

Reply:
I was just reading about this today.. present returns false for false, whitespace, and nil (opposite of blank). I initially had !nil, but present? seemed like it would still work given the use case is for a user_level.